src/stores/login.ts

```diff
@@ -13,16 +20,60 @@ export const emptyUser: User = {
 export const useLoginStore = defineStore('login', {
   state: () => ({
     user: emptyUser,
+    token: '',
   }),
 
   getters: {
     getUser: (state): User => state.user,
+    getToken: (state): string => state.token,
   },
 
   actions: {
     setUser(user: User): void {
       Object.assign(this.user, user);
     },
+    setToken(token: string): void {
+      this.token = token;
+    },
+    /**
+     * Login user
+     * Checks if email and password are set.
+     * If not, shows a notification.
+     * If yes, sends the login request to the API.
+     * If successful, sets the token.
+     * @returns Promise<LoginResponse | null>
+     */
+    async login(): Promise<LoginResponse | null> {
+      const { apiPost } = useApiPost();
+      // check that email is set
+      if (!this.user.email) {
+        Notify.create({
+          message: i18n.global.t('login.form.messageEmailReqired'),
+          color: 'negative',
+        });
+        return null;
+      }
+      // check that password is set
+      if (!this.user.password) {
+        Notify.create({
+          message: i18n.global.t('login.form.messagePasswordRequired'),
+          color: 'negative',
+        });
+        return null;
+      }
+      // login
+      const { data } = await apiPost<LoginResponse>({
+        endpoint: '/login',
```

Comment:
Don't hardcode URL endpoint string, please. Import URL endpoint from ./src/router/routes_conf.ts please.

Reply:
Updated. Defined in routes_conf.